packages/flutter/lib/src/material/dialog.dart

```diff
@@ -42,6 +42,7 @@ class Dialog extends StatelessWidget {
     this.child,
     this.insetAnimationDuration = const Duration(milliseconds: 100),
     this.insetAnimationCurve = Curves.decelerate,
+    this.shape,
```

Comment:
You can provide the default shape here.

Reply:
That would work better in this context, but in my follow up PR I'm going to create a `DialogTheme`. That means the default has to be set below because we will have something in `build` along the lines of 
```dart
shape: shape ?? themeData.dialogTheme.shape ?? _defaultDialogShape
```

Reply:
I see. So if I want a dialog that's just a rectangle: `shape: Border()` - does that work?.

Reply:
Yes, that works. For the sake of completeness I added a unit test for this case.